server/controllers/user.js

```diff
@@ -1,23 +1,155 @@
-const Todo = require('../models').Todo;
+import models from '../models';
+import jwt from 'jsonwebtoken';
+import config from '../config/config';
 
-module.exports = {
-  create(req, res) {
-    return Todo
-      .create({
-        title: req.body.title,
+const Users = models.User;
+const Roles = models.Role;
+
+ const isLoggedInUser = (userId, queryId) => {
+    if (parseInt(userId) === parseInt(queryId)) return true;
+  };
+
+const UserController = {
+   create(req, res) {
+    const userData = req.body;
+    Users.findOne({
+      where: {
+        $or: [
+          {email: req.body.email},
+          {username: req.body.username},
+        ]
+      }
+    })
+    .then((dbUser) => {
+      if (dbUser) { // dbUser.username || dbUser.email
+        return res.status(409).send({ message: 'An account with that email address exists' });
+      }
+      Roles.findOne({
+        where:{
+          title: 'WRITER'
+        }
+      }).then((role) =>
+        Users.create(Object.assign({}, userData, { roleId: role.id }))
+          .then((newUser) => {
+            const payload = {
+              email: newUser.email
+            };
+            const token = jwt.sign(payload, config.jwtSecret, { expiresIn: '24h' });
+            return res.status(201).send({
+              newUser,
+              message: 'User signup completed successfully',
+              token
+            });
+          })
+      )
+    })
+    .catch(error => res.status(500).send(error));
+  },
+
+  login(req, res) {
+    if (req.body.email && req.body.password ){
+      const { email } = req.body;
+      const query = { where: { email } };
+
+      Users.findOne(query)
+        .then((user) => {
+          if (!user) {
+            return res.status(404).send({ message: 'Your account does not exist'});
+          }
+          if (Users.isPassword(user.password, req.body.password)) {
+            const payload =  {
```

Comment:
Check how many things are getting "signed" with the create token. This line has id, roleId, and email. The create has just one (email)